File: reduce_json/filter.py
import json
import sys
import os

fin = open(sys.argv[1], 'r')
fout = open(os.path.join(sys.argv[2], os.path.basename(sys.argv[1])), 'w')

keep = sys.argv[3:len(sys.argv)]

print "reading data"

data = json.loads(fin.read())
fin.close()

smaller = []

for d in data[:10]:
    print str(d)

    small = {}

    for k in keep:
        small[k] = d[k]


    smaller.append(small)


print "writing data"

json.dump(smaller, fout)
fout.close()

File: reduce_json/filter.rb
require 'json'

fin = File.open(ARGV[0], 'r')
fout = File.open(File.join(ARGV[1], File.basename(fin)), 'w')

keep = ARGV.slice(2,ARGV.size)

puts 'reading data'
data = JSON.load(fin)
fin.close()

smaller = []

data.each_with_index do |d, i|
  if i % 1000 == 0
    puts "#{i}/#{data.size}"
  end

  small = {}

  keep.each do |k|
    small[k] = d[k]
  end

  smaller << small
end

puts 'writing to ' + File.path(fout)
JSON.dump(smaller, fout)
fout.close()
